src/vs/workbench/services/runtimeSession/common/runtimeSession.ts

```diff
@@ -794,6 +794,102 @@ export class RuntimeSessionService extends Disposable implements IRuntimeSession
 		return this._foregroundSession;
 	}
 
+	/**
+	 * Prompts the user to interrupt a session to complete another action.
+	 *
+	 * @param session The session to interrupt.
+	 * @param action The localized action name that will occur after interrupting the session.
+	 * @returns Whether the session was interrupted.
+	 */
+	private async promptToInterruptSession(session: ILanguageRuntimeSession, action: string): Promise<boolean> {
+		enum PromptOption {
+			Interrupt,
+			DoNotInterrupt,
+			TransitionToIdle
+		}
+
+		const activeSession = this._activeSessionsBySessionId.get(session.sessionId);
+		if (!activeSession) {
+			return false;
+		}
+
+		const disposables = new DisposableStore();
+
+		// If the runtime is busy, we need to interrupt it before restarting.
+		// But first, confirm with the user.
+		const promptResult = await new Promise<PromptOption>(resolve => {
+			let transitionedToIdle = false;
+
+			const notificationHandle = this._notificationService.prompt(
+				Severity.Warning,
+				localize('positron.console.interruptPrompt.confirm', 'The runtime is busy. Do you want to interrupt it and {0}? You\'ll lose any unsaved objects.', action),
+				[
+					{
+						label: localize('positron.runtime.restart.confirm.yes', 'Yes'),
```

Comment:
We should probably remove `restart` from the localization key since we use this for delete as well now!
```suggestion
						label: localize('positron.runtime.confirm.yes', 'Yes'),
```